### ISSUE TITLE
[MNT]: Unify maximize plot function for all backends

### ISSUE THREAD

**Issue by moschmdt** [matplotlib/matplotlib#21585]
### Summary

Generalize the function to maximize a plot window via the figure manager, e.g. `mng = plt.get_current_fig_manager()` so that the same function can be called independent of the backend used.

### Proposed fix

At the moment the command/ function to maximize a plot window is slightly different for every backend, cf. [Stackoverflow - first answer](https://stackoverflow.com/questions/12439588/how-to-maximize-a-plt-show-window-using-python). I suggest to add a function like the following (I have this function already in use at work):

```
import matplotlib

def maximize_plot(manager):
    backend = matplotlib.get_backend()
  
    if backend == "TkAgg":
        manager.window.state('zoomed')

    elif: backend == "QtAgg":
        manager.window.showMaximized()

    else:
        raise ValueError("Backend not available")
```

Obviously, not every backend is listed here but I think the general approach is clearly visible. This function should be part of the manager class I guess? 

- [ ]  Is anything like that already planned?
- [ ] Should it be implemented?
- [ ]  Where should it be implemented?

I would be happy to implement that and start contributing!

**Comment by timhoffm** [comment]
Thanks for the suggestion! I'm sceptical on replicating such GUI-related operations. That seems beyond the responsibility of matplotlib. Weighing usefulness vs. maintainance burden, I have concerns. OTOH the SO answer has 182 upvotes, so apparently at least some users want this.

If this gets implemented, it should be an API defined in `FigureManagerBase` and implemented in the backend-specific subclasses. See https://matplotlib.org/devdocs/api/backend_bases_api.html#matplotlib.backend_bases.FigureManagerBase.

**Comment by moschmdt** [comment]
Thanks for the hint to the specific code section. I had a look at it and will probably give it a shot on the weekend.

**Comment by anntzer** [comment]
I agree with @timhoffm that this seems out of matplotlib's scope; perhaps this could evolve into a separate "library gui-toolkit-independent methods" (and matplotlib's managers and canvases could just make sure that there's a uniform way to access the underlying window object, e.g. `.window`, etc.).

**Comment by timhoffm** [comment]
Closing as out of scope.
